frontend/src/components/Reports/routine/Nonconformity-Unit.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import config from "../../../config.json"; // Assuming config.json is in the same directory
+import {
+  DataTable,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  DatePicker,
+  Button, // Added Button component
+} from "@carbon/react";
+
+const NonconformityReportsByUnit = () => {
+  const [reports, setReports] = useState([]);
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [printableDataUrl, setPrintableDataUrl] = useState(null); // Added state for printable data URL
+
+  useEffect(() => {
+    fetchReports();
+    fetchPrintableDataUrl(); // Fetch the printable data URL when the component mounts
+  }, [startDate, endDate]);
+
+  const fetchReports = async () => {
+    try {
+      const url = `${config.serverBaseUrl}/api/reports`;
+      const params = new URLSearchParams({
+        type: "nonConformity",
+        report: "nonConformityByUnit",
+        lowerDateRange: startDate ? startDate.toISOString() : null,
+        upperDateRange: endDate ? endDate.toISOString() : null,
+      });
+      const response = await fetch(`${url}?${params}`);
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+      const data = await response.json();
+      setReports(data);
+    } catch (error) {
+      console.error("Error fetching data:", error);
+    }
+  };
+
+  const fetchPrintableDataUrl = async () => {
+    try {
+      const response = await fetch(`${config.serverBaseUrl}/api/getPrintableDataUrl`);
+      if (!response.ok) {
+        throw new Error("Failed to fetch printable data URL");
+      }
+      const url = await response.text();
+      setPrintableDataUrl(url);
+    } catch (error) {
+      console.error("Error fetching printable data URL:", error);
+    }
+  };
+
+  const handleGeneratePrintableData = async () => {
+    try {
+      const response = await fetch(printableDataUrl); // Use the printable data URL
+      if (!response.ok) {
+        throw new Error("Failed to generate printable data");
+      }
+      const printableData = await response.json();
+      // Handle the printable data, e.g., open in a new window or initiate download
+    } catch (error) {
+      console.error("Error generating printable data:", error);
+    }
+  };
+
+  return (
+    <div>
```

Comment:
And what will happen is you happen to have another div here ?

do you think you may need 
```
return(
  <>
      //your code here
  </>
);
```